### PR TITLE
add base layer.rs

### PR DESCRIPTION
**Pull request by hone**
Add the start to a `Layer` struct with core functionality. We'll still want to add an `add_layer` fn to `Build`.

### CODE REVIEW

File: src/data/layer.rs [hunk 1]
```diff
@@ -12,3 +12,14 @@ pub struct Layer {
     pub cache: bool,
     pub metadata: Table,
 }
+
+impl Layer {
+    pub fn new() -> Self {
+        Layer {
+            launch: defaults::r#false(),
+            build: defaults::r#false(),
+            cache: defaults::r#false(),
```
Comment (Malax) [inline src/data/layer.rs:21]: Shouldn't we just use `bool` literals there? I think we have those `fn`'s for serde which does not support defaults from literals.

File: src/error.rs [hunk 2]
```diff
@@ -12,4 +12,10 @@ pub enum Error {
         "Found `{0}` but value MUST only contain numbers, letters, and the characters ., /, and -."
     )]
     InvalidStackId(String),
+    #[error("could not serialize into TOML")]
+    TomlSerError(#[from] toml::ser::Error),
+    #[error("could not deserialize from TOML")]
+    TomlDeError(#[from] toml::de::Error),
+    #[error("I/O Error")]
+    IoError(#[from] std::io::Error),
```
Comment (Malax) [inline src/error.rs:20]: Will this make the `Display` of that `io::Error` appear somewhere? If not, that would complicate debugging.

**Review by Malax — APPROVED** [review]
👍🏻 , especially for all the rustdoc :)

File: src/data/layer.rs (hunk 1 above)
Comment (hone) [reply src/data/layer.rs:21]: yeah, I was thinking about that. I picked this in case we ever change the default but happy to switch to false.

**Pull request merged**